src/app/table/data-table/terra-data-table.component.ts

```diff
@@ -324,7 +324,7 @@ export class TerraDataTableComponent<T, P> implements OnInit, OnChanges
         {
             return arg
                    && arg.type && typeof arg.type == 'string'
-                   && arg.value && typeof arg.value == 'string';
+                   && arg.value && (typeof arg.value == 'string' || typeof arg.value == 'number' || typeof arg.value == 'function');
```

Comment:
In dem Fall solltest du auch im Interface für `value` nur Werte von diesen Typen zulassen.